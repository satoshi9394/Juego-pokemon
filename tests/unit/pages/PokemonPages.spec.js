import { shallowMount, mount } from '@vue/test-utils'
import PokemonPages from '@/pages/PokemonPage';
import PokemonOptions from '@/components/PokemonOptions';
import PokemonPicture from '@/components/PokemonPicture';


import { pokemons } from '../mocks/pokemons.mock'

describe('vista de PokemonPages', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonPages);
  });

  test('debe de hacer match con el shapshot ', () => {
    expect(wrapper.html()).toMatchSnapshot()
  });
  test('debe de llamar el mixPokemonArray al montar', () => {
    const mixPokemonArrayspy = jest.spyOn(PokemonPages.methods, 'mixPokemonArray');
    shallowMount(PokemonPages);
    expect(mixPokemonArrayspy).toHaveBeenCalled()
  });

  test('debe de hacer match con el snapshot cuando carga los pokemons', () => {
    const wrapper = shallowMount( PokemonPages, {
      data() {
        return {
          pokemonArr: pokemons,
          pokemon: pokemons[0],
          showPokemon: false,
          loader: false,
          showAnswer: false,
          message: ''
        }
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('debe de mostrar los componentes de pokemonPicture y PokemonOption', () => {
    const wrapper = shallowMount( PokemonPages, {
      data() {
        return {
          pokemonArr: pokemons,
          pokemon: pokemons[0],
          showPokemon: false,
          loader: false,
          showAnswer: false,
          message: ''
        }
      }
    });
    const PokemonOptionsBox = wrapper.findComponent(PokemonOptions);
    const PokemonPictureBox = wrapper.findComponent(PokemonPicture);
    expect(PokemonOptionsBox.exists()).toBeTruthy();
    expect(PokemonPictureBox.exists()).toBeTruthy();
    expect(PokemonPictureBox.attributes('pokemonid')).toBe('1');
    expect(PokemonOptionsBox.attributes('pokemons')).toBeTruthy();
  });

  test('Prueba de respuesta a la pregunta del componente ', async() => {
    const wrapper = shallowMount( PokemonPages, {
      data() {
        return {
          pokemonArr: pokemons,
          pokemon: pokemons[0],
          showPokemon: false,
          loader: false,
          showAnswer: false,
          message: ''
        }
      }
    });
    await wrapper.vm.checkAnswer(1);
    expect(wrapper.find('h2').exists()).toBeTruthy();
    expect(wrapper.vm.showPokemon).toBeTruthy();
    expect(wrapper.find('h2').text()).toBe(`Correcto, ${pokemons[0].name}`);

    await wrapper.vm.checkAnswer(2);
    expect(wrapper.find('h2').text()).toBe(`Oops, era ${pokemons[0].name}`);
  })
  
});