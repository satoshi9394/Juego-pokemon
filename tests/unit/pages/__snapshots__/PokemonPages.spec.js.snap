// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vista de PokemonPages debe de hacer match con el shapshot  1`] = `<loader-stub></loader-stub>`;

exports[`vista de PokemonPages debe de hacer match con el snapshot cuando carga los pokemons 1`] = `
<div class="fade-in">
  <h1> Quien es este pokemon? </h1>
  <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub>
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
  <!--v-if-->
</div>
`;
